Remove the need for jQuery

torotil opened the pull request.

torotil committed:
Switch to microbundle and update

torotil committed:
[jquery] $.isArray() → Array.isArray()

torotil committed:
Use native elements in storageKeys

torotil committed:
jquery: Replace $.attr() with getAttribute/setAttribute

torotil committed:
jquery: Replace $.filter() with Array.prototype.filter()

torotil committed:
jquery: Replace $.trigger() with dispatchEvent()

torotil committed:
jquery: Replace $.extend()

Note: $.extend(true, [], settings.stores) was rather weird to begin with
as it always created a copy of the stores.

torotil committed:
jquery: Replace $.closest() with element.closest()

torotil committed:
Remove usage of $.data() and return a WeakMap instead

torotil committed:
jquery: Replace $.contains()

torotil committed:
Remove $self: Use double-arrow functions and plain JS

torotil committed:
Remove $inputs

Switching to a native event listener means that we have to trigger
native events $.trigger() doesn’t do that, that’s why the tests need
changing as well.

torotil committed:
Replace $write

torotil committed:
Pass DOM-element to the Api class

torotil committed:
[refactor] Use element.dataset to access data attributes

torotil committed:
Reimplement jQuery.val()

torotil committed:
$.fn.formPrefill() into a small wrapper function

torotil committed:
Move bulk of the code out of the jQuery function wrapper

torotil committed:
[refactor] Turn Api into a real class

torotil committed:
Simplify read-value map handling

torotil committed:
Export functions

torotil committed:
Tests: Use imports instead of jQuery for tests

torotil committed:
Remove jQuery from the package name and docs

katzmo commented on README.md:
```diff
@@ -49,32 +53,34 @@ $('form').formPrefill({
 
 // Clear each field’s values from the stores, leave the current field values untouched:
-$('form').data('formPrefill').removeAll({resetFields: false});
+$('form').data('formPrefill').removeAll({resetFields: false})
```
Looks like some left over jQuery example?

torotil committed:
[refactor] Replace Array.prototype functions

The alternatives seem to be more concise and readable but might be
slightly slower.

torotil committed:
Remove the (now unused) privates object

torotil committed:
Change the formPrefill function so that it works one single element

Working on multiple wrappers can easily be done in the code using the
API and it seems to be the less common use-case.

torotil committed:
Use a global apiRegistry to expose individual elements

The formPrefill() call can now return the wrapper’s API object instead
of the entire registry. Interacting with individual elements is the less
common use-case and should warrant an explicit import.

torotil committed:
README.md: Add info on how to run the tests

torotil committed:
[refactor] Better comments and simplifications in val.js

torotil committed:
Prepare the package for release

torotil committed:
Reinstall eslint and reformat code

katzmo commented on src/Stores.js:
```diff
@@ -7,7 +7,7 @@ import { defaults } from './defaults'
 const $ = jQuery
```
Some left over jQuery, doesn’t look like `$` is still used.

katzmo commented on README.md:
```diff
@@ -46,35 +50,37 @@ $('form').formPrefill({
 // Prefill all fields that have values saved in the stores (this is done automatically when you call the plugin on a form):
-$('form').data('formPrefill').readAll();
+let form = document.querySelector('form')
+let api = formPrefill([form])
```
Doesn’t `formPrefill()` expect a single wrapperElement now (like in the example above)?

katzmo reviewed (changes requested):
I think the example.html needs some updates too.

There still is a lot of potential for more comments and better documentation but IMHO the code is already a lot easier to read than it was before :+1:

torotil committed:
Remove left-over reference to jQuery

torotil committed:
[fixup] README: Fixup formPrefill() call with an array argument

torotil committed:
[refactor] Simplify key-prefixing using Array.map()

torotil committed:
[refactor] Replace filter with :checked selector

torotil committed:
Port the example.html

torotil committed:
[fix] Fix mismatching event.detail for form-prefill:stores-initialized

torotil committed:
[lint] Include tests/ for linting

torotil committed:
v0.14.1
